project/src/com/company/services/CSVService.java

```diff
@@ -0,0 +1,169 @@
+package com.company.services;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class CSVService {
```

Comment:
it was required that services that hold generic methods for read/write should be singleton

Reply:
The CSVService class is now a singleton as well